src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMappingAdjournmentTest.java

```diff
@@ -0,0 +1,185 @@
+    void getHearingDurationAdjournmentReturnsNullWithFeatureFlagEnabled() {
+        given(referenceDataServiceHolder.getHearingDurations()).willReturn(hearingDurations);
+
+        given(hearingDurations.getHearingDuration(BENEFIT_CODE, ISSUE_CODE)).willReturn(null);
+
+        SscsCaseData caseData = SscsCaseData.builder()
+            .benefitCode(BENEFIT_CODE)
+            .issueCode(ISSUE_CODE)
+            .appeal(Appeal.builder()
+                .hearingOptions(HearingOptions.builder()
+                    .wantsToAttend("Yes")
+                    .build())
+                .build())
+            .build();
+
+        Integer durationAdjourned = HearingsDurationMapping.getHearingDurationAdjournment(
+            caseData, referenceDataServiceHolder);
+        assertThat(durationAdjourned).isNull();
+
+        Integer result = HearingsDurationMapping.getHearingDuration(
+            caseData,
+            referenceDataServiceHolder
+        );
+
+        assertThat(result).isEqualTo(HearingsDurationMapping.DURATION_DEFAULT);
+    }
+
+    @DisplayName("When a valid duration is given but adjournCaseDurationUnits is not provided "
+        + "getHearingDuration returns the default adjournment duration")
+    @Test
+    void getHearingDurationWithNullUnits() {
+        given(hearingDurations.getHearingDuration(BENEFIT_CODE, ISSUE_CODE))
+            .willReturn(new HearingDuration(
+                BenefitCode.PIP_NEW_CLAIM,
+                Issue.DD,
+                HearingsDurationMappingTest.DURATION_FACE_TO_FACE,
+                HearingsDurationMappingTest.DURATION_INTERPRETER,
+                HearingsDurationMappingTest.DURATION_PAPER
+            ));
+
+        given(referenceDataServiceHolder.getHearingDurations()).willReturn(hearingDurations);
+        setAdjournmentDurationAndUnits(2, null);
+
+        int result = HearingsDurationMapping.getHearingDuration(caseData, referenceDataServiceHolder);
+
+        assertThat(result).isEqualTo(HearingsDurationMappingTest.DURATION_PAPER);
+    }
+
+    @DisplayName("When an invalid adjournCaseDuration and valid adjournCaseDurationUnits is given "
+        + "getHearingDuration a null pointer exception is thrown")
+    @ParameterizedTest
+    @MethodSource
+    void getHearingDurationWithInvalidUnitsThrowsException(
+        Integer adjournCaseDuration,
+        AdjournCaseNextHearingDurationUnits adjournCaseDurationUnits
+    ) {
+        setAdjournmentDurationAndUnits(adjournCaseDuration, adjournCaseDurationUnits);
+
+        assertThatThrownBy(() -> HearingsDurationMapping.getHearingDuration(caseData, referenceDataServiceHolder))
+            .isInstanceOf(NullPointerException.class);
+    }
+
+    private Stream<Arguments> getHearingDurationWithInvalidUnitsThrowsException() {
+        return Stream.of(
+            Arguments.of(null, SESSIONS),
+            Arguments.of(0, SESSIONS),
+            Arguments.of(0, MINUTES),
+            Arguments.of(null, MINUTES)
+        );
+    }
```

Comment:
Can you instead split this into multiple tests using enumsource?

Reply:
done